src/insights_client/tests/test_motd.py

```diff
@@ -0,0 +1,116 @@
+import os
+import tempfile
+import unittest
+import unittest.mock
+
+import pytest
+
+import insights_client
+
+
+class MockFS:
+    """Mock filesystem.
+
+    Ensures that the files required for the MOTD manipulation are created
+    in a temporary directory instead of working with actual system files.
+    """
+
+    def __init__(self):
+        self.dir = tempfile.TemporaryDirectory()
+
+        # /etc/insights-client/insights-client.motd
+        self.dir_src = unittest.mock.patch(
+            "insights_client.MOTD_SRC", f"{self.dir.name}/src"
+        )
+        self.dir_src.start()
+
+        # /etc/motd.d/insights-client
+        self.dir_motd = unittest.mock.patch(
+            "insights_client.MOTD_FILE", f"{self.dir.name}/motd"
+        )
+        self.dir_motd.start()
+
+        # /etc/insights-client/.registered
+        self.dir_reg = unittest.mock.patch(
+            "insights_client.REGISTERED_FILE", f"{self.dir.name}/reg"
+        )
+        self.dir_reg.start()
+
+        # /etc/insights-client/.unregistered
+        self.dir_unreg = unittest.mock.patch(
+            "insights_client.UNREGISTERED_FILE", f"{self.dir.name}/unreg"
+        )
+        self.dir_unreg.start()
+
+        with open(f"{self.dir.name}/src", "w") as f:
+            f.write("This is the content of the MOTD file.")
+
+    def __del__(self):
+        self.dir_unreg.stop()
+        self.dir_reg.stop()
+        self.dir_motd.stop()
+        self.dir_src.stop()
+        self.dir = None
+
+
+@pytest.fixture
+def mock_fs():
+    fs = MockFS()
+    yield fs
+    del fs
```

Comment:
I can imagine doing this without the class. But as mentioned above, I like how it encapsulates the state and leverages the object deletion.

```suggestion
    fs = setup_fs()
    yield fs
    teardown_fs(fs)
```

Another option would be to put the function contents directly in here. With the suggested for-cycle, it’s not even that complicated piece of code to be inlined.